internal/gen/types/types.go

```diff
@@ -35,6 +35,11 @@ func Run(ctx context.Context, projectId string, dbConfig pgconn.Config, lang str
 	}
 	included := strings.Join(schemas, ",")
 
+	var defaultSchemaEnv string
+	if len(schemas) == 1 && schemas[0] != "public" {
+		defaultSchemaEnv = schemas[0]
+	}
```

Comment:
We should add a new flag so it doesn't break existing users that rely on the generated default schema.

Reply:
@sweatybridge Thanks. I've implemented a `set-default` flag for backwards compatibility that makes the behavior now explicit and opt-in. The new behavior is now:

```sh
supabase gen types --schema private --set-default
```